File: theater-service/src/main/java/com/example/theater_service/model/Theater.java

package com.example.theater_service.model;

import com.google.cloud.firestore.GeoPoint;
import com.google.cloud.firestore.annotation.DocumentId;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Theater {
    private String theaterId;
    private String name;
    private String address;
    private String city;
    private GeoPoint location;
}

File: AndroidApp/app/src/main/java/com/btlcnpm/androidapp/data/model/Theater.kt
package com.btlcnpm.androidapp.data.model

// DTO cho Theater (Rạp chiếu)
// Các tên trường này phải khớp với JSON trả về từ theater-service

/**
 * Model này đại diện cho GeoPoint của Firestore.
 * Khi Jackson (thư viện JSON) serialize `com.google.cloud.firestore.GeoPoint`
 * nó thường trả về một object có chứa `latitude` và `longitude`.
 */
data class GeoPoint(
    val latitude: Double = 0.0,
    val longitude: Double = 0.0
)

/**
 * Model đại diện cho một rạp chiếu phim (Theater).
 * Các trường đều là nullable (?) để đảm bảo an toàn khi parse JSON.
 */
data class Theater(
    val theaterId: String?,
    val name: String?,
    val address: String?,
    val city: String?,
    val location: GeoPoint? // Dùng model GeoPoint đã định nghĩa ở trên
)

